osu.Game/Screens/Edit/ExternalEditScreen.cs

```diff
@@ -0,0 +1,222 @@
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Platform;
+using osu.Framework.Screens;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Overlays;
+using osu.Game.Screens.OnlinePlay.Match.Components;
+using osuTK;
+
+namespace osu.Game.Screens.Edit
+{
+    internal partial class ExternalEditScreen : OsuScreen
+    {
+        private readonly Task<ExternalEditOperation<BeatmapSetInfo>> fileMountOperation;
+
+        [Resolved]
+        private GameHost gameHost { get; set; } = null!;
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        private readonly Editor editor;
+
+        public ExternalEditOperation<BeatmapSetInfo>? EditOperation;
+
+        private double timeLoaded;
+
+        private FillFlowContainer flow = null!;
+
+        public ExternalEditScreen(Task<ExternalEditOperation<BeatmapSetInfo>> fileMountOperation, Editor editor)
+        {
+            this.fileMountOperation = fileMountOperation;
+            this.editor = editor;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            InternalChild = new Container
+            {
+                Masking = true,
+                CornerRadius = 20,
+                AutoSizeAxes = Axes.Both,
+                AutoSizeDuration = 500,
+                AutoSizeEasing = Easing.OutQuint,
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Children = new Drawable[]
+                {
+                    new Box
+                    {
+                        Colour = colourProvider.Background5,
+                        RelativeSizeAxes = Axes.Both,
+                    },
+                    flow = new FillFlowContainer
+                    {
+                        Margin = new MarginPadding(20),
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Vertical,
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                        Spacing = new Vector2(15),
+                    }
+                }
+            };
+
+            showSpinner("Exporting for edit...");
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            timeLoaded = Time.Current;
+
+            fileMountOperation.ContinueWith(t =>
+            {
+                EditOperation = t.GetResultSafely();
```

Comment:
No error handling? This will throw (and presumably kill the game) if *any* I/O goes wrong in the setting up of the mounting (which it famously does all of the time, if you look at sentry).